packages/devtools_app/lib/src/memory/flutter/memory_heatmap.dart

```diff
@@ -0,0 +1,497 @@
+
+    // Add the external heap to the heat map.
+    root.children.putIfAbsent('External Heap', () {
+      final node = Node(
+        'External Heap',
+        parent: root,
+        children: <String, Node>{},
+      );
+      node.byteSize = snapshotGraph.externalSize;
+      return node;
+    });
+
+    // Add the filtered libraries/classes to the heat map.
+    root.children.putIfAbsent('All Filtered Libraries', () {
+      final node = Node(
+        'All Filtered Libraries',
+        parent: root,
+        children: <String, Node>{},
+      );
+      node.byteSize = snapshotGraph.shallowSize - root.byteSize;
+      return node;
+    });
+
+    root.byteSize = snapshotGraph.shallowSize + snapshotGraph.externalSize;
+
+    return InstructionsSize(root);
+  }
+
+  final Node root;
+}
+
+class Node {
+  Node(
+    this.name, {
+    this.byteSize = 0,
+    this.parent,
+    this.children = const <String, Node>{},
+  })  : assert(name != null),
+        assert(byteSize != null),
+        assert(children != null);
+
+  final String name;
+  int byteSize;
+  Rect rect = Rect.zero;
+  final Node parent;
+  final Map<String, Node> children;
+
+  Iterable<Node> get ancestors {
+    final nodes = <Node>[];
+    Node current = this;
+    while (current.parent != null) {
+      nodes.add(current.parent);
+      current = current.parent;
+    }
+    return nodes;
+  }
+
+  bool get isLeaf => children.isEmpty;
+
+  Node findRect(Offset offset) {
+    if (rect.contains(offset)) {
+      return this;
+    }
+    for (var ancestor in ancestors) {
+      if (ancestor.rect.contains(offset)) {
+        return ancestor;
+      }
+    }
+    for (var child in children.values) {
+      final value = child.findRect(offset);
+      if (value != null) {
+        return value;
+      }
+    }
+    return null;
+  }
+
+  @override
+  String toString() => 'Node($name, $byteSize, $rect)';
+}
+
+class Symbol {
+  const Symbol({
+    @required this.name,
+    @required this.size,
+    this.libraryUri,
+    this.className,
+  })  : assert(name != null),
+        assert(size != null);
+
+  static Symbol fromMap(Map<String, dynamic> json) {
+    return Symbol(
+      name: json['n'] as String,
+      size: json['s'] as int,
+      className: json['c'] as String,
+      libraryUri: json['l'] as String,
+    );
+  }
+
+  final String name;
+  final int size;
+  final String libraryUri;
+  final String className;
+
+  List<String> get parts {
+    return <String>[
+      if (libraryUri != null) ...libraryUri.split('/') else '@stubs',
+      if (className != null && className.isNotEmpty) className,
+      name,
+    ];
+  }
+}
+
+class HeatMapSizeAnalyzer extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      color: Colors.blueGrey,
+      home: FlameChart(InstructionsSize.fromSnapshop(heapGraph)),
+      debugShowCheckedModeBanner: false,
+    );
+
+    // TODO(terry): Testing search needs to be stateful.
```

Comment:
remove this commented out code or implement

Reply:
Implemented.